### PR TITLE
settings(feat): reset to default value

### PR DESCRIPTION
**Pull request by chipsTM**
adds #576 

image

taking suggestions on css formatting and handling custom components/ignores

### CODE REVIEW

**Comment by chipsTM** [comment]
I was thinking that another property (`disableReset` ?) should be added to `SettingNode` interface so that a setting can be declared as non-resettable when one uses `declareConfig`. At least this way we can exclude custom types or types with non-primitive values?

**Comment by Melonify** [comment]
The behavior of this isn't quite right. Ideally it should remove the setting from the stored table entirely, so that if a setting's default value changes for some reason it's as if the user had never changed the setting.

Allowing the post-reset state to be separate from the initial state adds to potential user confusion.

File: src/app/settings/SettingsNode.vue [hunk 1]
```diff
@@ -50,6 +58,14 @@ function onHover(): void {
 	useTimeoutFn(() => emit("seen"), 500);
 }
 
+const currentSetting = useConfig<SevenTV.SettingType>(props.node.key);
+
+// set the currentSetting back to default to trigger UI change before removing from the settings db
+function resetSetting() {
+	currentSetting.value = props.node.defaultValue;
+	db.settings.delete(props.node.key);
```
Comment (AnatoleAM) [inline src/app/settings/SettingsNode.vue:66]: Please handle errors with `catch()` on the promise

**Review by AnatoleAM — CHANGES_REQUESTED** [review]

**Review by AnatoleAM — APPROVED** [review]
Thank you for the contribution :slightly_smiling_face:

**Pull request merged**